GFW/convert.py

```diff
@@ -0,0 +1,170 @@
+edate = datetime.datetime(2050, 1, 1, 0, 0, 0)
+proj = 'EPSG:4326'
+github_path = 'https://github.com/rubisco-sfa/ILAMB-Data/blob/msteckler/GFW/convert.py'
+
+# data sources
+remote_data = 'https://www.arcgis.com/sharing/rest/content/items/f950ea7878e143258a495daddea90cc0/data'
+local_data = 'sequestration_rate_mean_aboveground_full_extent_Mg_C_ha_yr.tif'
+remote_u_data = 'https://www.arcgis.com/sharing/rest/content/items/d28470313b8e443aa90d5cbcd0f74163/data'
+local_u_data = 'sequestration_error_ratio_layer_in_full_extent.tif'
+
+# function to open a raster and set the CRS
+def open_raster(local_data, remote_data):
+    if not os.path.isfile(local_data):
+        urlretrieve(remote_data, local_data)
+    download_stamp = time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(os.path.getmtime(local_data)))
+    data = rxr.open_rasterio(local_data, band_as_variable=True)
+    
+    # ensure data is not projected and is wgs84
+    epsg_code = int(data.rio.crs.to_epsg())
+    if epsg_code != 4326:
+        data = data.rio.reproject(crs=proj)
+
+    return data, download_stamp
+
+# open data and error files
+data, download_stamp = open_raster(local_data, remote_data)
+errd, _ = open_raster(local_u_data, remote_u_data)
+
+# function to resample to 0.5 degrees
+def coarsen(target_res, data):
+    resampled_data = data.coarsen(x=(int(target_res / abs(data.rio.resolution()[0]))),
+                                  y=(int(target_res / abs(data.rio.resolution()[1])))).mean()
+    return resampled_data
+
+# resample
+data = coarsen(0.5, data)
+errd = coarsen(0.5, errd)
+
+# rename dims
+ds = data.rename({'x': 'lon', 'y': 'lat', 'band_1':'fVeg'})
+er = errd.rename({'x': 'lon', 'y': 'lat', 'band_1':'error_ratio'})
+# convert error ratio to standard error
+er['standard_error'] = er['error_ratio'] * ds['fVeg']
+
+# numpy array of time bounds
+tb_arr = np.asarray([
+    [cf.DatetimeNoLeap(sdate.year, sdate.month, sdate.day)],
+    [cf.DatetimeNoLeap(edate.year, edate.month, edate.day)]
+]).T
+
+# xarray data array of time bounds
+tb_da = xr.DataArray(tb_arr, dims=('time', 'nv'))
+
+# add time dimension and time bounds attribute
+ds = ds.expand_dims(time=tb_da.mean(dim='nv'))
+ds['time_bounds'] = tb_da
+er = er.expand_dims(time=tb_da.mean(dim='nv'))
+er['time_bounds'] = tb_da
+
+# add standard error variable
+ds = ds.assign(standard_error=er['standard_error'])
+
+# https://cfconventions.org/Data/cf-documents/requirements-recommendations/conformance-1.8.html
+# edit time attributes
+t_attrs = {
+    'axis':'T',
+    'long_name':'time'}
+
+# edit lat/lon attributes
+y_attrs = {
+    'axis':'Y',
+    'long_name':'latitude',
+    'units':'degrees_north'}
+x_attrs = {
+    'axis':'X',
+    'long_name':'longitude',
+    'units':'degrees_east'}
+
+# edit variable attributes
+v_attrs = {
+    'long_name': 'mean aboveground biomass carbon accumulation rate',
+    'units': 'Mg ha-1 yr-1',
+    'ancillary_variables': 'fVeg standard_error'}
```

Comment:
Unless I misunderstand something, the `fVeg` doesn't belong here. That is the variable name, these should refer to variable names of related variables and in this case is only `standard_error`.

Reply:
Yes, you're right! `'fVeg standard_error'` should be the long name attribute for the error variable. Good eye!

From the CF conventions site:
```  
float q(time) ;
    q:standard_name = "specific_humidity" ;
    q:units = "g/g" ;
    q:ancillary_variables = "q_error_limit q_detection_limit" ;
  float q_error_limit(time)
    q_error_limit:standard_name = "specific_humidity standard_error" ;
    q_error_limit:units = "g/g" ;
```